media_driver/linux/common/os/i915/mos_bufmgr.c

```diff
@@ -4431,8 +4431,194 @@ mos_gem_bo_is_exec_object_async(struct mos_linux_bo *bo)
+    INTEL_HWCONFIG_L3_ALLOC_PER_BANK_UNIFIED_TILE_CACHE,    /* 56 */
+    INTEL_HWCONFIG_L3_ALLOC_PER_BANK_COMMAND_BUFFER,    /* 57 */
+    INTEL_HWCONFIG_L3_ALLOC_PER_BANK_RW,            /* 58 */
+    INTEL_HWCONFIG_MAX_NUM_L3_CONFIGS,          /* 59 */
+    INTEL_HWCONFIG_BINDLESS_SURFACE_OFFSET_BIT_COUNT,   /* 60 */
+    INTEL_HWCONFIG_RESERVED_CCS_WAYS,           /* 61 */
+    INTEL_HWCONFIG_CSR_SIZE_IN_MB,              /* 62 */
+    INTEL_HWCONFIG_GEOMETRY_PIPES_PER_SLICE,        /* 63 */
+    INTEL_HWCONFIG_L3_BANK_SIZE_IN_KB,          /* 64 */
+    INTEL_HWCONFIG_SLM_SIZE_PER_DSS,            /* 65 */
+    INTEL_HWCONFIG_MAX_PIXEL_FILL_RATE_PER_SLICE,       /* 66 */
+    INTEL_HWCONFIG_MAX_PIXEL_FILL_RATE_PER_DSS,     /* 67 */
+    INTEL_HWCONFIG_URB_SIZE_PER_SLICE_IN_KB,        /* 68 */
+    INTEL_HWCONFIG_URB_SIZE_PER_L3_BANK_COUNT_IN_KB,    /* 69 */
+    INTEL_HWCONFIG_MAX_SUBSLICE,                /* 70 */
+    INTEL_HWCONFIG_MAX_EU_PER_SUBSLICE,         /* 71 */
+    INTEL_HWCONFIG_RAMBO_L3_BANK_SIZE_IN_KB,        /* 72 */
+    INTEL_HWCONFIG_SLM_SIZE_PER_SS_IN_KB,           /* 73 */
+    __INTEL_HWCONFIG_LIMIT
+};
+
+enum {
+    INTEL_HWCONFIG_MEMORY_TYPE_LPDDR4 = 0,
+    INTEL_HWCONFIG_MEMORY_TYPE_LPDDR5,
+};
+
+#define BIT(x) (1ul <<(x))
+enum {
+    INTEL_HWCONFIG_CACHE_TYPE_L3    = BIT(0),
+    INTEL_HWCONFIG_CACHE_TYPE_LLC   = BIT(1),
+    INTEL_HWCONFIG_CACHE_TYPE_EDRAM = BIT(2),
+};
+
+#ifndef DRM_I915_QUERY_HWCONFIG_TABLE
+#define DRM_I915_QUERY_HWCONFIG_TABLE   5
+#endif
+
+static int mos_query_items(int fd, struct drm_i915_query_item *query_item)
+{
+    struct drm_i915_query query;
+    int ret;
+
+    assert(query_item);
+    memclear(query);
+    query.num_items = 1;
+    query.items_ptr = (uintptr_t)query_item;
+
+    return drmIoctl(fd, DRM_IOCTL_I915_QUERY, &query);
+}
 
 int mos_query_device_blob(int fd, MEDIA_SYSTEM_INFO* gfx_info)
 {
-    return -1;
+    uint32_t *hw_info;
+    uint32_t  ulength= 0;
+    int ret, i;
+    struct drm_i915_query_item query_item;
+
+    hw_info = nullptr;
+
+    memclear(query_item);
+    query_item.length = 0;
+    query_item.query_id = DRM_I915_QUERY_HWCONFIG_TABLE;
+    ret = mos_query_items(fd, &query_item);
+    if (ret != 0 || query_item.length <= 0)
+    {
+        return (ret != 0) ? ret : -1;
+    }
+
+    hw_info = (uint32_t*) malloc(query_item.length);
+    if (hw_info != nullptr)
+        memset(hw_info, 0, query_item.length);
+    else
+        return -ENOMEM;
+    query_item.data_ptr = (uintptr_t) hw_info;
+    ret = mos_query_items(fd, &query_item);
+    if (ret != 0 || query_item.length <= 0)
+    {
+        return (ret != 0) ? ret : -1;
+    }
+    ulength = query_item.length / sizeof(uint32_t);
+    i = 0;
+    assert(gfx_info);
+    while (i < ulength) {
+        /* Attribute ID starts with 1 */
+        assert(hw_info[i] > 0);
+
+        if (INTEL_HWCONFIG_MAX_SLICES_SUPPORTED == hw_info[i])
+        {
+            assert(hw_info[i+1] == 1);
```

Comment:
about these assert, the information size will never change in future for each configure?

Reply:
The number of bytes used by the MAX_SLICES_SUPPORTED field is fixed and cannot ever change. The value contained within those bytes will (potentially) be different on every platform.

Also, why use a long series of 'if' statements instead of a 'switch' statement?

Reply:
Agree, switch is better. I have updated the patch. Thanks.